lib/cacheServerClient.ts

```diff
@@ -73,38 +73,36 @@ export class CacheServerClient {
       );
     });
     const token = await retry.execute(async () => {
-      const [address, blockNumber, chainId] = await Promise.all([
-        this.signer.getAddress(),
-        this.provider.getBlockNumber(),
-        this.signer.getChainId(),
-      ]);
-
-      const chainName = knownChains[chainId];
-      this.chainName = chainName || 'volta';
-
-      const { arrayify, keccak256 } = utils;
-      const { encodedHeader, encodedPayload } =
-        this.createLoginTokenHeadersAndPayload({
-          address,
-          blockNumber,
-          chainName,
-        });
-      const msg = `0x${Buffer.from(
-        `${encodedHeader}.${encodedPayload}`
-      ).toString('hex')}`;
-      const sig = await this.signer.signMessage(arrayify(keccak256(msg)));
-      const encodedSignature = base64url(sig);
-      const { data } = await this.httpClient.post<{ token: string }>('/login', {
-        identityToken: `${encodedHeader}.${encodedPayload}.${encodedSignature}`,
-      });
+      const {
+        data: { nonce },
+      } = await this.httpClient.post<{ nonce: string }>('/login/siwe/initiate');
+      const uri = new URL('/v1/login/siwe/verify', new URL(this.url).origin)
+        .href;
+      const message = new SiweMessage({
+        nonce,
+        domain: new URL(this.url).host,
+        address: await this.signer.getAddress(),
+        uri,
+        version: '1',
+        chainId: await this.signer.getChainId(),
+      }).prepareMessage();
+
+      const signature = await this.signer.signMessage(message);
+
+      const { data } = await this.httpClient.post<{ token: string }>(
+        '/login/siwe/verify',
+        {
+          message,
+          signature,
+        }
+      );
```

Comment:
I'd add a debug log line in case of error. It will reduce pain for someone who will troubleshoot :)

Reply:
I think the error should be logged already in retry https://github.com/energywebfoundation/passport-did-auth/blob/d17bd1fc3b6bc581b34acb775e34f76830d9f858/lib/cacheServerClient.ts#L60